iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/schemaregion/mtree/impl/pbtree/CachedMTreeStore.java

```diff
@@ -771,9 +698,8 @@ public void skipTemplateChildren() {
     @Override
     public void close() {
       try {
-        if (nextNode != null) {
-          unPin(nextNode, false);
-          nextNode = null;
+        if (mergeIterator.hasNext()) {
+          unPin(mergeIterator.next(), false);
```

Comment:
Since there's no buffered result in this mergeIterator, close it directly without reading the next element.